File: Day6.cs
using System.Text.RegularExpressions;

class Day6
{
    static void Main()
    {
        string[] data = File.ReadLines("input.txt").ToArray();
        Console.WriteLine("Part 1 Answer: " + Part1(data));
        Console.WriteLine("Part 2 Answer: " + Part2(data));
    }

    static int Part1(string[] data)
    {
        int product = 1;
        int[] times = Array.ConvertAll(Regex.Replace(data[0], @"\s+", " ").Substring(6).Split(" "), int.Parse);
        int[] distances = Array.ConvertAll(Regex.Replace(data[1], @"\s+", " ").Substring(10).Split(" "), int.Parse);
        for (int i = 0; i < times.Length; i++)
        {
            double dicriminant = Math.Sqrt(Math.Pow(times[i], 2) - 4 * distances[i]);
            int lower = (int)Math.Floor((float)((times[i] - dicriminant) / 2));
            int upper = (int)Math.Ceiling((float)((times[i] + dicriminant) / 2));
            product *= upper - lower - 1;
        }
        return product;
    }

    static long Part2(string[] data)
    {
        long time = long.Parse(Regex.Replace(data[0], @"\s+", "").Substring(5));
        long distance = long.Parse(Regex.Replace(data[1], @"\s+", "").Substring(9));
        double dicriminant = Math.Sqrt(Math.Pow(time, 2) - 4 * distance);
        long lower = (long)Math.Floor((float)((time - dicriminant) / 2));
        long upper = (long)Math.Ceiling((float)((time + dicriminant) / 2));
        return upper - lower - 1;
    }
}

File: 2025/Day6.js
function part1(input) {
    const data = input.split("\n").map(
        r => r.trim().replace( /\s\s+/g, ' ' ).split(" ")
    );
    let total = 0;
    for (let i = 0; i < data[0].length; i++) {
        let op = data[data.length - 1][i];
        let answer = op === "+" ? 0 : 1;
        for (let j = 0; j < data.length - 1; j++) {
            num = parseInt(data[j][i]);
            answer = op === "+" ? answer + num : answer * num;
        }
        total += answer;
    }
    return total;
}

function part2(input) {
    const data = input.split("\n").map(r => r.split(""));
    let total = 0;
    let op = data[data.length - 1][0];
    let answer = op === "+" ? 0 : 1;
    for (let i = 0; i < data[0].length; i++) {
        let num = 0;
        for (let j = 0; j < data.length - 1; j++) {
            if (data[j][i] !== " ") {
                num = num * 10 + parseInt(data[j][i]);
            }
        }
        if (num === 0) {
            total += answer;
            op = data[data.length - 1][i + 1];
            answer = op === "+" ? 0 : 1;
            continue;
        }
        answer = op === "+" ? answer + num : answer * num;
    }
    total += answer;
    return total;
}

const fs = require('fs');
fs.readFile('input.txt', 'utf8', (err, input) => {
    if (err) {
        console.error('Error reading input:', err);
        return;
    }
    console.log(`Part 1 Answer: ${part1(input)}`);
    console.log(`Part 2 Answer: ${part2(input)}`);
});